exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestParquetReaderDecision.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import org.apache.drill.common.expression.SchemaPath;
+import org.apache.drill.exec.store.parquet.AbstractParquetScanBatchCreator;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.hadoop.ParquetFileReader;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+
+/**
+ * DRILL-5797 introduces more granularity on new reader use cases. This test is aimed at
+ * checking correctness of function used for new reader usage decision making.
+ */
+public class TestParquetReaderDecision {
+
+  private static final String path = "src/test/resources/store/parquet/complex/complex.parquet";
+  private static Configuration conf;
+  private static ParquetMetadata footer;
+
+  @BeforeClass
+  public static void setUpBeforeClass() throws Exception {
+    conf = new Configuration();
+
+    try {
+      footer = ParquetFileReader.readFooter(conf, new Path(path));
+    } catch (IOException ioe) {
+      fail("Could not read Parquet file '" + path + "', error message: " + ioe.getMessage()
+          + " cwd: " + Paths.get(".").toAbsolutePath().normalize().toString());
+      throw(ioe);
+    }
+  }
+
+  @Test
+  public void testParquetReaderDecision() {
```

Comment:
Please split into several unit tests, unit tests names should show what cases they are testing.

Reply:
I will fix this and all other test issues you highlighted. When fixing them I realised that existing `complex.parquet` test file possibly doesn't provide desired coverage for changes in this PR. Bulk part of changes are aimed at making utility functions work with schemas like 
`a`
`b`.`a`
while schema in `complex.parquet` doesn't contain such corner cases.
Does it make sense to add new resource to cover such cases and base tests on that file?

Reply:
Yes, just please generate small file to cover all needed cases.

Reply:
Done, schema is:
```
file schema:   root
--------------------------------------------------------------------------------
id:            OPTIONAL INT64 R:0 D:1
a:             OPTIONAL BINARY O:UTF8 R:0 D:1
repeated:      REPEATED INT64 R:1 D:1
VariableCase:  OPTIONAL BINARY O:UTF8 R:0 D:1
nested:        OPTIONAL F:3
.id:           OPTIONAL INT64 R:0 D:2
.repeated:     REPEATED INT64 R:1 D:2
.VaRiAbLeCaSe: OPTIONAL BINARY O:UTF8 R:0 D:2
```